sql/catalyst/src/main/scala/org/apache/spark/sql/catalog/v2/LookupCatalog.scala

```diff
@@ -99,6 +99,21 @@ trait LookupCatalog extends Logging {
     }
   }
 
+  object CatalogAndRestNameParts {
+    def unapply(nameParts: Seq[String]): Some[(Option[CatalogPlugin], Seq[String])] = {
+      if (nameParts.isEmpty) {
+        Some((defaultCatalog, Nil))
```

Comment:
Shouldn't we introduce `currentCatalog` in `CatalogManager` and use it instead of `defaultCatalog` as @rdblue suggested? I was thinking doing something like the following: https://github.com/apache/spark/compare/master...imback82:current_ns

Reply:
I'd like to make this PR a pure refactor and leave it to followups. For example, we should add more tests when changing this to `currentCatalog`.

Reply:
Got it.